File: devops-scm-spring-boot-starter/src/main/java/com/tencent/devops/scm/spring/manager/NoSuchScmProviderException.java


package com.tencent.devops.scm.spring.manager;

import static java.util.Objects.requireNonNull;

import com.tencent.devops.scm.api.exception.ScmApiException;
import lombok.Getter;

@Getter
public class NoSuchScmProviderException extends ScmApiException {

    private final String providerName;

    public NoSuchScmProviderException(String providerName) {
        super("No such provider: '" + providerName + "'.");
        this.providerName = requireNonNull(providerName);
    }
}

File: devops-scm-spring-boot-starter/src/main/kotlin/com/tencent/devops/scm/spring/manager/NoSuchScmProviderException.kt
package com.tencent.devops.scm.spring.manager

import com.tencent.devops.scm.api.exception.ScmApiException
import java.util.Objects.requireNonNull

class NoSuchScmProviderException(private val providerName: String) :
    ScmApiException("No such provider: '$providerName'.") {
    
    init {
        requireNonNull(providerName)
    }
}
